src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml

```diff
@@ -0,0 +1,101 @@
+    <item name="Error"   value="0x03"/>
+  </enum>
+
+  <enum name="ErrorStateEnum" type="ENUM8">
+    <cluster code="0x0060"/>
+    <item name="NoError"                    value="0x00"/>
+    <item name="UnableToStartOrResume"      value="0x01"/>
+    <item name="UnableToCompleteOperation"  value="0x02"/>
+    <item name="CommandInvalidInState"      value="0x03"/>
+  </enum>
+
+  <struct name="OperationalStateStruct">
+    <cluster code="0x0060"/>
+    <item name="OperationalStateID"    type="OperationalStateEnum" optional="false"/>
+    <item name="OperationalStateLabel" type="CHAR_STRING" optional="false"/>
+  </struct>
+
+  <struct name="ErrorStateStruct">
+    <cluster code="0x0060"/>
+    <item name="ErrorStateID" type="ErrorStateEnum" optional="false" />
+    <item name="ErrorStateLabel" type="CHAR_STRING" optional="false" length="64"/>
+    <item name="ErrorStateDetails" type="CHAR_STRING" optional="false" length="64"/>
+  </struct>
+
+  <cluster>
+    <domain>General</domain>
+    <name>Operational State</name>
+    <code>0x0060</code>
+    <define>OPERATIONAL_STATE_CLUSTER</define>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <description>This cluster supports remotely monitoring and, where supported, changing the operational state of any device where a state machine is a part of the operation.</description>
+
+    <globalAttribute side="server" code="0xFFFD" value="1" />
+
+    <attribute side="server" code="0x0000" define="PHASE_LIST"              type="ARRAY" entryType="CHAR_STRING"                                     writable="false" isNullable="true"  optional="false">PhaseList</attribute>
+    <attribute side="server" code="0x0001" define="CURRENT_PHASE"           type="INT8U"                                    min="0x00" max="0x1F"    writable="false" isNullable="true"  optional="false">CurrentPhase</attribute>
+    <attribute side="server" code="0x0002" define="COUNTDOWN_TIME"          type="INT32U"                                min="0x00" max="0x3F480" writable="false" isNullable="true"  optional="true">CountdownTime</attribute>
+    <attribute side="server" code="0x0003" define="OPERATIONAL_STATE_LIST"  type="ARRAY" entryType="OperationalStateStruct"                          writable="false"                    optional="false">OperationalStateList</attribute>
+    <attribute side="server" code="0x0004" define="OPERATIONAL_STATE"       type="OperationalStateStruct"                                            writable="false"                    optional="false">OperationalState</attribute>
+    <attribute side="server" code="0x0005" define="OPERATIONAL_ERROR"       type="ErrorStateStruct"                                                   writable="false"                    optional="false">OperationalError</attribute>
+
+    <command source="client" code="0x00" name="Pause" response="OperationalCommandResponse" optional="false">
```

Comment:
All the commands here are marked optional in the spec, no?

Reply:
It seems that OPERATIONAL_STATE_LIST、OPERATIONAL_STATE、OPERATIONAL_ERROR are mandatory from https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/OperationalState.adoc